sdk/clientcore/annotation-processor-test/pom.xml

```diff
@@ -78,25 +78,65 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.13.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
-        <configuration>
-          <proc>only</proc>
-          <!-- Supported source version 'RELEASE_8' from annotation processor 'io.clientcore.annotation.processor.AnnotationProcessor' less than -source '17' -->
-          <failOnWarning>false</failOnWarning>
-          <generatedSourcesDirectory>${project.build.directory}/generated-sources/</generatedSourcesDirectory>
-          <annotationProcessorPaths>
-            <annotationProcessorPath>
-              <groupId>io.clientcore</groupId>
-              <artifactId>annotation-processor</artifactId>
-              <version>1.0.0-beta.1</version> <!-- {x-version-update;io.clientcore:annotation-processor;current} -->
-            </annotationProcessorPath>
-          </annotationProcessorPaths>
-          <annotationProcessors>
-            <annotationProcessor>io.clientcore.annotation.processor.AnnotationProcessor</annotationProcessor>
-          </annotationProcessors>
-          <compileSourceRoots>
-            <compileSourceRoot>${project.build.directory}/generated-sources/</compileSourceRoot>
-          </compileSourceRoots>
-        </configuration>
+        <executions>
+          <execution>
+            <id>run-annotation-processing</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+
+            <configuration>
+              <source>1.8</source>
+              <target>1.8</target>
+              <release>8</release>
```

Comment:
Didn't we add this to handle the Javapoet SourceVersion issue. Do we still want to keep this?

Reply:
This was added as annotation processing can raise a javac warning when the version declared by the processor is lower than the version of Java being used to run the processor. Which is external to Java Poet or JavaParser and will need to be kept.